src/filesystem/search_id.rs

```diff
@@ -0,0 +1,25 @@
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt-log", derive(defmt::Format))]
+/// Unique ID used to search for files and directories in the open File/Directory lists
+pub struct SearchId(pub(crate) u32);
+
+/// ID generator intented to be used in a static context.
+///
+/// This object will always return a different ID.
+pub struct IdGenerator {
+    next_id: u32,
```

Comment:
I know this is going away, but if this was a WrappingU32, you wouldn't need to remember to call .wrapping_add on it.